ui/welcome/src/commonMain/kotlin/social/androiddev/welcome/composables/WelcomeScreen.kt

```diff
@@ -1,50 +1,136 @@
+//import androidx.compose.desktop.ui.tooling.preview.Preview
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
 import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.layout.fillMaxWidth
 import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.aspectRatio
 import androidx.compose.foundation.layout.height
-import androidx.compose.material.Button
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.widthIn
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.MaterialTheme
 import androidx.compose.material.Surface
 import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
+import social.androiddev.common.composables.buttons.MastodonButton
+import social.androiddev.common.composables.buttons.MastodonOutlinedButton
+import social.androiddev.common.theme.Blue
+import social.androiddev.common.theme.MastodonTheme
+import social.androiddev.common.utils.AsyncImage
+import social.androiddev.common.utils.loadImageIntoPainter
 
 @Composable
 fun WelcomeScreen(
     modifier: Modifier = Modifier,
+    navigateToLogin: () -> Unit,
+    navigateToSignUp: () -> Unit,
+    appIcon: @Composable () -> Unit = {
+        AsyncImage(
+            load = { loadImageIntoPainter(url = "https://picsum.photos/200") },
```

Comment:
Is it fine to have this image here? license?
If not, switch to a placeholder: https://via.placeholder.com/200x200/6FA4DE/010041?text=MastodonX

Reply:
Yes - open source random image generator. It's not static. And will be replaced shortly with a real app asset/image